src/connection.cc

```diff
@@ -85,6 +100,34 @@ bool Connection::handle_read(const boost::system::error_code& error,
   return true; // success
 }
 
+std::string Connection::shortenPath(std::string path) {
+  // Reached last possible path.
+  if (path == "/") {
+    return "";
+  }
+
+  // Reverse the path to find the first instance of '/'
+  std::string reversed(path);
+  std::reverse(reversed.begin(), reversed.end());
+  std::string::size_type pos = reversed.find('/');
```

Comment:
instead of manually reversing the string and using `string::find`, you can use the reverse find function `string::rfind` instead to get the position of the last "/"
http://www.cplusplus.com/reference/string/string/rfind/